arangod/Agency/FailedLeader.cpp

```diff
@@ -309,13 +326,22 @@ bool FailedLeader::start(bool& aborts) {
         addPreconditionServerHealth(pending, _to, "GOOD");
         // Server list in plan still as before
         addPreconditionUnchanged(pending, planPath, planned);
-        // Server list in Current still as known
-        addPreconditionUnchanged(pending, curPath + "/servers", current);
-        // Failover candidates in Current still as known
-        auto const& failoverCandidates = _snapshot.hasAsSlice(curPath + "/failoverCandidates");
-        if (failoverCandidates.second) {
-          addPreconditionUnchanged(pending, curPath + "/failoverCandidates", failoverCandidates.first);
-        }
+        // Check that Current/servers and failoverCandidates are still as
+        // we inspected them:
+        doForAllShards(_snapshot, _database, shardsLikeMe,
+            [this, &pending](Slice plan, Slice current,
+                             std::string& planPath,
+                             std::string& curPath) {
+              addPreconditionUnchanged(pending, curPath, current);
+              // take off "servers" from curPath and add
+              // "failoverCandidates":
+              std::string foCandsPath = curPath.substr(0, curPath.size() - 7);
+              foCandsPath += StaticStrings::FailoverCandidates;
+              auto foCands = this->_snapshot.hasAsSlice(foCandsPath);
+              if (foCands.second) {
+                addPreconditionUnchanged(pending, foCandsPath, foCands.first);
```

Comment:
same here, don't we need to add a precondition in case there have been no failoverCandidates yet?